atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java

```diff
@@ -175,36 +178,7 @@ public Transaction commitAndStartNewTransaction(Transaction tx) {
 
     @Override
     public Transaction createNewTransaction() {
-        long startTimestamp = timelockService.getFreshTimestamp();
-        PathTypeTracker pathTypeTracker = PathTypeTrackers.constructSynchronousTracker();
-        return transactionWrapper.apply(
-                new SnapshotTransaction(
-                        metricsManager,
-                        keyValueServiceWrapper.apply(keyValueService, pathTypeTracker),
-                        timelockService,
-                        lockWatchManager,
-                        transactionService,
-                        NoOpCleaner.INSTANCE,
-                        () -> startTimestamp,
-                        getConflictDetectionManager(),
-                        SweepStrategyManagers.createDefault(keyValueService),
-                        startTimestamp,
-                        Optional.empty(),
-                        PreCommitConditions.NO_OP,
-                        AtlasDbConstraintCheckingMode.NO_CONSTRAINT_CHECKING,
-                        null,
-                        TransactionReadSentinelBehavior.THROW_EXCEPTION,
-                        false,
-                        timestampValidationReadCache,
-                        getRangesExecutor,
-                        defaultGetRangesConcurrency,
-                        sweepQueueWriter,
-                        deleteExecutor,
-                        validateLocksOnReads,
-                        () -> TRANSACTION_CONFIG,
-                        ConflictTracer.NO_OP,
-                        tableLevelMetricsController),
-                pathTypeTracker);
+        return Iterables.getOnlyElement(startTransactions(ImmutableList.of(PreCommitConditions.NO_OP)));
```

Comment:
this could almost be a default method on the `TestTransactionManager` interface (but fine here too).

Reply:
Also we could do something similar to `commitAndStartNewTransaction`.